From 47de887d7658cfd68df44b3acf710971054f957b Mon Sep 17 00:00:00 2001
From: Christian Persch <chpe@src.gnome.org>
Date: Fri, 26 Mar 2021 18:17:25 +0100
Subject: [PATCH 1/2] glib: Use stock glib macro to define boxed type

Poppler already depends on a sufficiently new glib version, so we can
simply use G_DEFINE_BOXED_TYPE instead of defining our own macro.
---
 glib/poppler-action.cc            |  4 ++--
 glib/poppler-annot.cc             |  2 +-
 glib/poppler-document.cc          |  6 +++---
 glib/poppler-page.cc              | 20 ++++++++++----------
 glib/poppler-private.h            | 15 ---------------
 glib/poppler-structure-element.cc |  4 ++--
 6 files changed, 18 insertions(+), 33 deletions(-)

diff --git glib/poppler-action.cc glib/poppler-action.cc
index d395db797..b772a3c2d 100644
--- glib/poppler-action.cc
+++ glib/poppler-action.cc
@@ -25,7 +25,7 @@
  * @title: PopplerAction
  */

-POPPLER_DEFINE_BOXED_TYPE (PopplerDest, poppler_dest, poppler_dest_copy, poppler_dest_free)
+G_DEFINE_BOXED_TYPE(PopplerDest, poppler_dest, poppler_dest_copy, poppler_dest_free)

 /**
  * poppler_dest_copy:
@@ -88,7 +88,7 @@ static PopplerActionLayer *poppler_action_layer_copy(PopplerActionLayer *action_
     return retval;
 }

-POPPLER_DEFINE_BOXED_TYPE (PopplerAction, poppler_action, poppler_action_copy, poppler_action_free)
+G_DEFINE_BOXED_TYPE(PopplerAction, poppler_action, poppler_action_copy, poppler_action_free)

 /**
  * poppler_action_free:
diff --git glib/poppler-annot.cc glib/poppler-annot.cc
index f4c5bb1fc..53205658f 100644
--- glib/poppler-annot.cc
+++ glib/poppler-annot.cc
@@ -1779,9 +1779,7 @@
 }

 /* PopplerAnnotCalloutLine */
-POPPLER_DEFINE_BOXED_TYPE (PopplerAnnotCalloutLine, poppler_annot_callout_line,
-			   poppler_annot_callout_line_copy,
-			   poppler_annot_callout_line_free)
+G_DEFINE_BOXED_TYPE(PopplerAnnotCalloutLine, poppler_annot_callout_line, poppler_annot_callout_line_copy, poppler_annot_callout_line_free)

 /**
  * poppler_annot_callout_line_new:
diff --git glib/poppler-document.cc glib/poppler-document.cc
index e8dfe1379..eafe9b63e 100644
--- glib/poppler-document.cc
+++ glib/poppler-document.cc
@@ -2426,9 +2426,7 @@
 };


-POPPLER_DEFINE_BOXED_TYPE (PopplerIndexIter, poppler_index_iter,
-			   poppler_index_iter_copy,
-			   poppler_index_iter_free)
+G_DEFINE_BOXED_TYPE(PopplerIndexIter, poppler_index_iter, poppler_index_iter_copy, poppler_index_iter_free)

 /**
  * poppler_index_iter_copy:
@@ -2663,9 +2661,7 @@
 	int index;
 };

-POPPLER_DEFINE_BOXED_TYPE (PopplerFontsIter, poppler_fonts_iter,
-			   poppler_fonts_iter_copy,
-			   poppler_fonts_iter_free)
+G_DEFINE_BOXED_TYPE(PopplerFontsIter, poppler_fonts_iter, poppler_fonts_iter_copy, poppler_fonts_iter_free)

 /**
  * poppler_fonts_iter_get_full_name:
@@ -3237,9 +3233,7 @@
   int index;
 };

-POPPLER_DEFINE_BOXED_TYPE (PopplerLayersIter, poppler_layers_iter,
-			   poppler_layers_iter_copy,
-			   poppler_layers_iter_free)
+G_DEFINE_BOXED_TYPE(PopplerLayersIter, poppler_layers_iter, poppler_layers_iter_copy, poppler_layers_iter_free)

 /**
  * poppler_layers_iter_copy:

diff --git glib/poppler-page.cc glib/poppler-page.cc
index 75c39c123..3332a9ebc 100644
--- glib/poppler-page.cc
+++ glib/poppler-page.cc
@@ -1549,9 +1549,7 @@

 /* PopplerRectangle type */

-POPPLER_DEFINE_BOXED_TYPE (PopplerRectangle, poppler_rectangle,
-			   poppler_rectangle_copy,
-			   poppler_rectangle_free)
+G_DEFINE_BOXED_TYPE(PopplerRectangle, poppler_rectangle, poppler_rectangle_copy, poppler_rectangle_free)

 /**
  * poppler_rectangle_new:
@@ -1596,9 +1594,7 @@

 /* PopplerPoint type */

-POPPLER_DEFINE_BOXED_TYPE (PopplerPoint, poppler_point,
-                           poppler_point_copy,
-                           poppler_point_free)
+G_DEFINE_BOXED_TYPE(PopplerPoint, poppler_point, poppler_point_copy, poppler_point_free)

 /**
  * poppler_point_new:
@@ -1650,9 +1646,7 @@

 /* PopplerQuadrilateral type */

-POPPLER_DEFINE_BOXED_TYPE (PopplerQuadrilateral, poppler_quadrilateral,
-                           poppler_quadrilateral_copy,
-                           poppler_quadrilateral_free)
+G_DEFINE_BOXED_TYPE(PopplerQuadrilateral, poppler_quadrilateral, poppler_quadrilateral_copy, poppler_quadrilateral_free)

 /**
  * poppler_quadrilateral_new:
@@ -1703,9 +1697,7 @@

 /* PopplerTextAttributes type */

-POPPLER_DEFINE_BOXED_TYPE (PopplerTextAttributes, poppler_text_attributes,
-			   poppler_text_attributes_copy,
-			   poppler_text_attributes_free)
+G_DEFINE_BOXED_TYPE(PopplerTextAttributes, poppler_text_attributes, poppler_text_attributes_copy, poppler_text_attributes_free)

 /**
  * poppler_text_attributes_new:
@@ -1809,7 +1801,7 @@
  */

 /* PopplerColor type */
-POPPLER_DEFINE_BOXED_TYPE (PopplerColor, poppler_color, poppler_color_copy, poppler_color_free)
+G_DEFINE_BOXED_TYPE(PopplerColor, poppler_color, poppler_color_copy, poppler_color_free)

 /**
  * poppler_color_new:
@@ -1856,9 +1848,7 @@
 }

 /* PopplerLinkMapping type */
-POPPLER_DEFINE_BOXED_TYPE (PopplerLinkMapping, poppler_link_mapping,
-			   poppler_link_mapping_copy,
-			   poppler_link_mapping_free)
+G_DEFINE_BOXED_TYPE(PopplerLinkMapping, poppler_link_mapping, poppler_link_mapping_copy, poppler_link_mapping_free)

 /**
  * poppler_link_mapping_new:
@@ -1913,9 +1903,7 @@
 }

 /* Poppler Image mapping type */
-POPPLER_DEFINE_BOXED_TYPE (PopplerImageMapping, poppler_image_mapping,
-			   poppler_image_mapping_copy,
-			   poppler_image_mapping_free)
+G_DEFINE_BOXED_TYPE(PopplerImageMapping, poppler_image_mapping, poppler_image_mapping_copy, poppler_image_mapping_free)

 /**
  * poppler_image_mapping_new:
@@ -1957,9 +1945,7 @@
 }

 /* Page Transition */
-POPPLER_DEFINE_BOXED_TYPE (PopplerPageTransition, poppler_page_transition,
-			   poppler_page_transition_copy,
-			   poppler_page_transition_free)
+G_DEFINE_BOXED_TYPE(PopplerPageTransition, poppler_page_transition, poppler_page_transition_copy, poppler_page_transition_free)

 /**
  * poppler_page_transition_new:
@@ -2006,9 +1992,7 @@
 }

 /* Form Field Mapping Type */
-POPPLER_DEFINE_BOXED_TYPE (PopplerFormFieldMapping, poppler_form_field_mapping,
-			   poppler_form_field_mapping_copy,
-			   poppler_form_field_mapping_free)
+G_DEFINE_BOXED_TYPE(PopplerFormFieldMapping, poppler_form_field_mapping, poppler_form_field_mapping_copy, poppler_form_field_mapping_free)

 /**
  * poppler_form_field_mapping_new:
@@ -2063,9 +2047,7 @@
 }

 /* PopplerAnnot Mapping Type */
-POPPLER_DEFINE_BOXED_TYPE (PopplerAnnotMapping, poppler_annot_mapping,
-			   poppler_annot_mapping_copy,
-			   poppler_annot_mapping_free)
+G_DEFINE_BOXED_TYPE(PopplerAnnotMapping, poppler_annot_mapping, poppler_annot_mapping_copy, poppler_annot_mapping_free)

 /**
  * poppler_annot_mapping_new:
diff --git glib/poppler-private.h glib/poppler-private.h
index d41b59c6a..102727162 100644
--- glib/poppler-private.h
+++ glib/poppler-private.h
@@ -146,25 +146,6 @@
 gboolean _poppler_convert_pdf_date_to_gtime (const GooString *date,
 					     time_t    *gdate);

-/*
- * A convenience macro for boxed type implementations, which defines a
- * type_name_get_type() function registering the boxed type.
- */
-#define POPPLER_DEFINE_BOXED_TYPE(TypeName, type_name, copy_func, free_func)          \
-GType                                                                                 \
-type_name##_get_type (void)                                                           \
-{                                                                                     \
-        static volatile gsize g_define_type_id__volatile = 0;                         \
-	if (g_once_init_enter (&g_define_type_id__volatile)) {                        \
-	        GType g_define_type_id =                                              \
-		    g_boxed_type_register_static (g_intern_static_string (#TypeName), \
-		                                  (GBoxedCopyFunc) copy_func,         \
-						  (GBoxedFreeFunc) free_func);        \
-		g_once_init_leave (&g_define_type_id__volatile, g_define_type_id);    \
-	}                                                                             \
-	return g_define_type_id__volatile;                                            \
-}
-
 void _poppler_error_cb (ErrorCategory category, Goffset pos, const char *message);

 #endif
diff --git glib/poppler-structure-element.cc glib/poppler-structure-element.cc
index aec7cd151..29d3e5cad 100644
--- glib/poppler-structure-element.cc
+++ glib/poppler-structure-element.cc
@@ -702,10 +702,7 @@
   unsigned index;
 };

-POPPLER_DEFINE_BOXED_TYPE (PopplerStructureElementIter,
-                           poppler_structure_element_iter,
-                           poppler_structure_element_iter_copy,
-                           poppler_structure_element_iter_free)
+G_DEFINE_BOXED_TYPE(PopplerStructureElementIter, poppler_structure_element_iter, poppler_structure_element_iter_copy, poppler_structure_element_iter_free)

 /**
  * poppler_structure_element_iter_copy:
@@ -900,10 +897,7 @@
   PopplerColor color;
 };

-POPPLER_DEFINE_BOXED_TYPE (PopplerTextSpan,
-                           poppler_text_span,
-                           poppler_text_span_copy,
-                           poppler_text_span_free)
+G_DEFINE_BOXED_TYPE(PopplerTextSpan, poppler_text_span, poppler_text_span_copy, poppler_text_span_free)

 enum {
   POPPLER_TEXT_SPAN_FIXED_WIDTH = (1 << 0),
--
GitLab


From bdd110b45a38e8a4f80f522892e4c4a9e432abd5 Mon Sep 17 00:00:00 2001
From: Christian Persch <chpe@src.gnome.org>
Date: Fri, 26 Mar 2021 18:17:25 +0100
Subject: [PATCH 2/2] glib: Remove incorrecly used volatile from enum type
 registration code

---
 glib/poppler-enums.c.template | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git glib/poppler-enums.c.template glib/poppler-enums.c.template
index 26a51b4c4..eefc76970 100644
--- glib/poppler-enums.c.template
+++ glib/poppler-enums.c.template
@@ -15,9 +15,9 @@
 GType
 @enum_name@_get_type (void)
 {
-  static volatile gsize g_define_type_id__volatile = 0;
+  static volatile gsize g_define_type_id = 0;

-  if (g_once_init_enter (&g_define_type_id__volatile)) {
+  if (g_once_init_enter (&g_define_type_id)) {
     static const G@Type@Value values[] = {
 /*** END value-header ***/

@@ -28,13 +28,13 @@ GType
 /*** BEGIN value-tail ***/
       { 0, NULL, NULL }
     };
-    GType g_define_type_id =
+    GType type =
        g_@type@_register_static (g_intern_static_string ("@EnumName@"), values);

-    g_once_init_leave (&g_define_type_id__volatile, g_define_type_id);
+    g_once_init_leave (&g_define_type_id, type);
   }

-  return g_define_type_id__volatile;
+  return g_define_type_id;
 }

 /*** END value-tail ***/
--
GitLab
